### PR TITLE
Add support for viewing Properties data, add 1MB value requirement

### PR DESCRIPTION
**Pull request by nickbp**
May be optionally supported by individual frameworks (by either providing or omitting the PropertyDeserializer). Adds a new /properties endpoint to StateResource to match.

Also:
- Refrain from throwing an error when listing or clearing Properties that aren't available (mirroring current task list/clear behavior)
- Set some noisy logs as 'debug'

### CODE REVIEW

File: src/main/java/org/apache/mesos/state/CuratorStateStore.java [hunk 1]
```diff
@@ -310,21 +311,29 @@ public void storeProperty(final String key, final byte[] value) throws StateStor
...
 
     @Override
-    public Collection<String> listPropertyKeys() throws StateStoreException {
+    public Collection<String> fetchPropertyKeys() throws StateStoreException {
         try {
             return curator.getChildren(this.propertiesPath);
+        } catch (KeeperException.NoNodeException e) {
+            // Root path doesn't exist yet. Treat as an empty list of properties. This scenario is
+            // expected to commonly occur when the Framework is being run for the first time.
+            return Collections.emptyList();
```
Comment (mohitsoni) [inline src/main/java/org/apache/mesos/state/CuratorStateStore.java]: Thanks for doing this!

File: src/main/java/org/apache/mesos/state/StateStoreUtils.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    /**
+     * Shared implementation for validating property value limits, for use by all StateStore
+     * implementations.
+     *
+     * @see StateStore#storeProperty(String, byte[])
+     */
+    public static void validateValue(byte[] value) throws StateStoreException {
+        final int maxLength = 1024 * 1024;
+        if (value.length > maxLength) {
```
Comment (mohitsoni) [inline src/main/java/org/apache/mesos/state/StateStoreUtils.java]: Validate `value != null` ?

File: src/main/java/org/apache/mesos/state/api/StateResource.java [hunk 3]
```diff
@@ -24,9 +24,28 @@
...
+    private final PropertyDeserializer propertyDeserializer;
 
+    /**
+     * Creates a new StateResource which cannot deserialize Properties.
+     *
+     * @param stateStore the source of data to be returned to users
+     */
     public StateResource(StateStore stateStore) {
         this.stateStore = stateStore;
+        this.propertyDeserializer = null;
```
Comment (mohitsoni) [inline src/main/java/org/apache/mesos/state/api/StateResource.java]: Refactor this constructor as `this(stateStore, null)` ?

File: src/test/java/org/apache/mesos/state/CuratorStateStoreTest.java [hunk 4]
```diff
@@ -406,41 +407,64 @@ public void testStoreInfoThenStatus() throws Exception {
...
     public void testPropertiesSlashStore() {
-        store.storeProperty(BAD_PROPERTY_KEY, PROPERTY_VALUE.getBytes(StandardCharsets.UTF_8));
+        store.storeProperty(SLASH_PROPERTY_KEY, PROPERTY_VALUE.getBytes(StandardCharsets.UTF_8));
     }
 
     @Test(expected = StateStoreException.class)
     public void testPropertiesSlashFetch() {
-        store.fetchProperty(BAD_PROPERTY_KEY);
+        store.fetchProperty(SLASH_PROPERTY_KEY);
     }
```
Comment (mohitsoni) [inline src/test/java/org/apache/mesos/state/CuratorStateStoreTest.java]: Add test for validating `null` value ?

Comment (nickbp) [inline src/test/java/org/apache/mesos/state/CuratorStateStoreTest.java]: Done, also reordered the property tests a bit into groups

File: src/main/java/org/apache/mesos/state/api/StateResource.java (hunk 3 above)
Comment (nickbp) [inline src/main/java/org/apache/mesos/state/api/StateResource.java]: Good point, Done

File: src/main/java/org/apache/mesos/state/StateStoreUtils.java (hunk 2 above)
Comment (nickbp) [inline src/main/java/org/apache/mesos/state/StateStoreUtils.java]: Added, Done

**Comment by mohitsoni** [comment]
LGTM now.
